### ISSUE TITLE
Error during Process Manager module compilation.

### ISSUE THREAD

**Issue by marcoonroad** [commanded/commanded#130]
I'm using the following Elixir environment:
```
$ elixir --version
Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

Elixir 1.6.0 (compiled with OTP 19)
```

And the following commanded version:
```
$ mix deps
...
* commanded 0.15.1 (https://github.com/commanded/commanded.git) (mix)
  locked at 720001e
...
```

I have the following error during compilation:
```Compiling 106 files (.ex)

== Compilation error in file lib/core_banking/transfer/internal_transfer/sagas/internal_transfer_saga.ex ==
** (UndefinedFunctionError) function :elixir_errors.warn/1 is undefined or private. Did you mean one of:

      * warn/3

    (elixir) :elixir_errors.warn("Commanded Deprecation Warning:\nProcess manager Elixir.CoreBanking.InternalTransfer.Sagas.InternalTransferSaga defined error/4 callback.\nThis is deprecated in favor of error/3\nSee https://github.com/commanded/commanded/blob/master/guides/Process%20Managers.md#deprecated-error4")
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
    /home/marcoonroad/Documents/Projects/Stone/ex-bank/lib/core_banking/transfer/internal_transfer/sagas/internal_transfer_saga.ex:1: Commanded.ProcessManagers.ProcessManager.__before_compile__/1
    (elixir) lib/kernel/parallel_compiler.ex:198: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/6
```

**Comment by marcoonroad** [comment]
Oh, never mind. This kind of error is due the usage of `error/4` instead of `error/3`. I'm closing this issue so. Sorry by the inconvenience.

**Comment by bamorim** [comment]
Actually, this is huge. It should not fail but only emit a warning. I implemented this using elixir 1.5.3 but haven't tested with 1.6 yet.

@slashdotdash can you open this until I make a PR with that?
